core/src/test/java/org/infinispan/eviction/impl/EvictionFunctionalTest.java

```diff
@@ -91,6 +91,22 @@ public void testSimpleEvictionMaxEntries() throws Exception {
       assertTrue(evictionListener.evictedEntries.size() > CACHE_SIZE);
    }
 
+   public void testEvictNonExistantEntry() {
+      String key = "key";
+      String value = "some-value";
+      cache.put(key, value);
+
+      cache.evict(key);
+
+      assertEquals(1, evictionListener.evictedEntries.size());
+
+      // Make sure if we evict again that it doesn't increase count
+      cache.evict(key);
+
+      // TODO: this seems like a bug, but many tests rely on this - maybe change later
```

Comment:
I'm sure if we had many tests that evicted an entry twice we'd have noticed the NPE earlier :)

Reply:
Yeah but this comment is the fact that an evict causes an event even when there as nothing to evict. :(